File: BOJ/11724.cpp
// 11724번_연결 요소의 개수
#include <iostream>
#include <vector>
#define MAX 1000
using namespace std;
vector<int> g[MAX];
bool visited[MAX];
void dfs(int u){
    visited[u] = true;
    for(int v=0; v<g[u].size(); ++v){
        if(!visited[g[u][v]])
            dfs(g[u][v]);
    }
}
int main(){
    int n, m, u, v, cnt=0;
    cin >> n >> m;
    for(int i=0; i<m; ++i){
        cin >> u >> v;
        g[v].push_back(u);
        g[u].push_back(v);
    }
    for(int u=1; u<=n; ++u){
        if(!visited[u]){
            dfs(u);
            cnt++;
        }
    }
    cout << cnt;
    return 0;
}

File: BOJ/11724.py
# 11724번_연결 요소의 개수
import sys
input = sys.stdin.readline
n, m = map(int, input().split())
a = [[] for _ in range(n+1)] # 각 노드와 연결된 노드를 저장하는 리스트
visited = [False] * (n+1) # 방문 배열

def dfs(v):
    visited[v] = True
    for i in a[v]: # 노드 v와 연결된 노드들 중 방문하지 않는 노드로 진행
        if not visited[i]:
            dfs(i)

for _ in range(m):
    u, v = map(int, input().split())
    a[u].append(v)
    a[v].append(u)

cnt = 0
for i in range(1, n+1):
    if not visited[i]:
        dfs(i) # i번째 노드에서 DFS시작
        cnt += 1 # DFS 한번 돌면 연결요소 하나를 탐색한 것

print(cnt)